src/modules/auth/api/auth.mutations.controllers.ts

```diff
@@ -1,4 +1,27 @@
-import { Controller } from "@nestjs/common";
+import { Body, Controller, Inject, Post, UsePipes, ValidationPipe } from "@nestjs/common";
+import { Paths } from "src/core/api/routes";
+import { LoginInput } from "src/modules/auth/api/auth.mutations.inputs";
+import { ApiAuth } from "src/modules/auth/api/auth.types";
+import { AuthServiceDefinition, AuthServiceName } from "src/modules/auth/domain/auth.service.types";
 
-@Controller("auth-mutations")
-export class AuthMutationsController {}
+@Controller(Paths.auth.root)
+@UsePipes(new ValidationPipe())
+export class AuthMutationsController {
+	constructor(@Inject(AuthServiceName) private authService: AuthServiceDefinition) {}
+
+	@Post(Paths.auth.mutations.login)
+	async login(@Body() input: LoginInput): Promise<ApiAuth> {
+		const serviceUser = await this.authService.login({
+			username: input.username,
+			password: input.password,
+		});
```

Comment:
This isn't how the login process should work. The login process should require the following

1. Fetch the user you're trying to log in with
2. Check the passwords. The user you fetch should have the same password as what's being sent in through the input
3. If the passwords don't match, you throw an error
4. If the passwords DO match, you can just return the user you initially fetched.


For this initial implementation of logging in, lets not worry about hashing. Just get the logic right before we do hashing.